### ISSUE TITLE
Unable to make REST API Request using AMAZON_COGNITO_USER_POOLS auth from Client Library

### ISSUE THREAD

**Issue by vmo-bstock** [aws-amplify/amplify-swift#3206]
### Describe the bug

When performing an REST API Request using the AWS Client Library, I am only able to receive back a "401 - Unauthorized" error. 

Even manually adding an "Authorization" header with the correct token returns a 401 Unauthorized response, when the exact same toke in Postman will work fine.  Additionally, if I intentionally omit the token in Postman, I get a 401 response.  If I intentionally provide an invalid token, I get a 403 response.  

This is leading my to believe that the "Authorization" header is actually not even being sent with the request.  If I try to manually add the header, same behavior, a 401 response.  If I provide an incorrect token to the Authorization header, a 401 response (whereby I would expect a 403 response in this situation)

Everything appears to be working correctly with the amplify configuration, with the exception of the REST API.  The GraphQL API and appsync/datastore work and the AUTH works as expected. 

This API was not configured with the Amplify CLI, as it is an existing resource.

Here is the relevant code that makes the request.  It is triggered by the Amplify Hub listener for "Signed In" event.
```
        do {
            authController = AuthController()
            Amplify.Hub.listen(to: .auth) { payload in
                switch payload.eventName {
                case HubPayload.EventName.Auth.signedIn:
                    Task.init {
                        do {
                            print("Hub says user signed in")
                            print("Token \(DeviceTokenManager.shared.deviceToken)")
                            try await registerDevice(token: DeviceTokenManager.shared.deviceToken)
                        } catch {
                            print("Failed to register device: \(error)")
                        }
                    }
                default:
                    break
                }
            }
            
        }
```

 And then this is the API Request:

```
    let request = RESTRequest(apiName: "myApi",  path: ApiEndpoints.DEVICE_REGISTRATION,  body: requestBody.data(using: .utf8) )

    do {
        let data = try await Amplify.API.post(request: request)
        let str = String(decoding: data, as: UTF8.self)
        print(">>DEVICE REGISTRATION SUCCESS: \(str)")
    } catch let error as APIError {
        print(">>DEVICE REGISTRATION ERROR: Failed due to API error: ", error)
    } catch {
        print(">>DEVICE REGISTRATION ERROR: Unexpected error: \(error.localizedDescription)")
    }
```
Here is the amplify configuration."  We build the configuration on the fly based on the response from a config server on initial app launch.

```
// Configure Amplify Plugins
    let apiPlugin = AWSAPIPlugin(modelRegistration: AmplifyModels())
    let dataStorePlugin = AWSDataStorePlugin(
        modelRegistration: AmplifyModels(),
        configuration: .custom(
            errorHandler: {error in Amplify.Logging.error(error: error) },

            //Sync Configuration
            syncInterval: .hours(24), // the maximum interval (in seconds) for which the system will continue to perform delta queries. After this, will perform full query.
            syncExpressions: syncExpressions
        )
    )
    let authPlugin = AWSCognitoAuthPlugin()
    let configuration = AmplifyConfiguration(

        api: APICategoryConfiguration(
            plugins: [
                "awsAPIPlugin": [
                    "myGQLApi": [
                        "endpointType": "GraphQL",
                        "endpoint": "https://<redacted>appsync-api.ca-central-1.amazonaws.com/graphql",
                        "region": "ca-central-1",
                        "authorizationType": "AMAZON_COGNITO_USER_POOLS"
                    ],
                    "myRestApi": [
                        "endpointType": "REST",
                        "endpoint": .string(config.API.endpoints[1].endpoint + "/ocmgr/"), 
                        "region": .string(config.API.endpoints[1].region),
                        "authorizationType": "AMAZON_COGNITO_USER_POOLS",
                    ]
                ]
            ]
        ),

        auth: AuthCategoryConfiguration(
            plugins: [
                "awsCognitoAuthPlugin": [
                    "UserAgent": "aws-amplify/cli",
                    "Version": "0.1.0",
                    "IdentityManager": [
                        "Default": []
                    ],
                    "AppSync": [
                        "Default": [
                            "ApiUrl": "https://<redacted>.appsync-api.ca-central-1.amazonaws.com/graphql",
                            "Region": "ca-central-1",
                            "AuthMode": "AMAZON_COGNITO_USER_POOLS",
                            "ClientDatabasePrefix": "myapp_AMAZON_COGNITO_USER_POOLS"                        ]
                    ],
                    "myapp_AWS_IAM": [
                        "ApiUrl": "https://<redacted>.appsync-api.ca-central-1.amazonaws.com/graphql",
                        "Region": "ca-central-1",
                        "AuthMode": "AWS_IAM",
                        "ClientDatabasePrefix": "myapp_AWS_IAM"
                    ],
                    "CredentialsProvider": [
                        "CognitoIdentity": [
                            "Default": [
                                "PoolId": .string(config.Auth.identityPoolId),
                                "Region": .string(config.Auth.region)
                            ]
                        ]
                    ],
                    "CognitoUserPool": [
                        "Default": [
                            "PoolId": .string(config.Auth.userPoolId),
                            "AppClientId": .string(config.Auth.userPoolWebClientId),
                            "Region": .string(config.Auth.region)
                        ]
                    ],
                    "Auth": [
                        "Default": [
                            "authenticationFlowType": "USER_SRP_AUTH",
                            "OAuth": [
                                "WebDomain": .string(config.Auth.oauth.domain),
                                "AppClientId":.string(config.Auth.userPoolWebClientId),
                                "SignInRedirectURI": .string(config.Auth.oauth.redirectSignIn),
                                "SignOutRedirectURI": .string(config.Auth.oauth.redirectSignOut),
                                "Scopes": .array(config.Auth.oauth.scope.map { .string($0) })
                            ]
                        ]
                    ],
                ]
            ]
        )
    )
    
    
    
    do {
//        Amplify.Logging.logLevel = .verbose // Set log level to verbose
        try Amplify.add(plugin: apiPlugin)
        try Amplify.add(plugin: dataStorePlugin)
        try Amplify.add(plugin: authPlugin)
        try Amplify.configure(configuration)
        print( "Amplify Initialized")
        
    } catch {
        print("Could not initialize Amplify: \(error)")
    }
    return config
}
```

### Steps To Reproduce

```swift
Easy, just use the code pasted above.
```

### Expected behavior

I would expect a valid 200 response back from the API Request. 

### Amplify Framework Version

2.17.0

### Amplify Categories

API, Auth

### Dependency manager

Swift PM

### Swift version

5

### CLI version

12.2.3

### Xcode version

14.3.1

### Relevant log output

```shell
<details>
<summary>Log Messages</summary>


2023-09-03 14:42:36.530304-0500 My App[60565:14036316] [Amplify] Adding plugin: <AWSAPIPlugin.AWSAPIPlugin: 0x600000b38a80>)
2023-09-03 14:42:36.531938-0500 My App[60565:14036316] [Amplify] Adding plugin: AWSDataStorePlugin.AWSDataStorePlugin)
2023-09-03 14:42:36.533971-0500 My App[60565:14036316] [Amplify] Adding plugin: AWSCognitoAuthPlugin.AWSCognitoAuthPlugin)
2023-09-03 14:42:36.534777-0500 My App[60565:14036316] [Amplify] Configuring
2023-09-03 14:42:36.539938-0500 My App[60565:14036316] [Amplify] Configuration: Optional(Amplify.AmplifyConfiguration(analytics: nil, api: Optional(Amplify.APICategoryConfiguration(plugins: ["awsAPIPlugin": Amplify.JSONValue.object(["MyAppApi": Amplify.JSONValue.object(["endpoint": Amplify.JSONValue.string("https://myApiUrl.com/ocmgr/"), "region": Amplify.JSONValue.string("ca-central-1"), "authorizationType": Amplify.JSONValue.string("AMAZON_COGNITO_USER_POOLS"), "endpointType": Amplify.JSONValue.string("REST")]), "MyAppManagerGQLApi": Amplify.JSONValue.object(["endpoint": Amplify.JSONValue.string("https://<redacted>.appsync-api.ca-central-1.amazonaws.com/graphql"), "region": Amplify.JSONValue.string("ca-central-1"), "endpointType": Amplify.JSONValue.string("GraphQL"), "authorizationType": Amplify.JSONValue.string("AMAZON_COGNITO_USER_POOLS")])])])), auth: Optional(Amplify.AuthCategoryConfiguration(plugins: ["awsCognitoAuthPlugin": Amplify.JSONValue.object(["CognitoUserPool": Amplify.JSONValue.object(["Default": Amplify.JSONValue.object(["Region": Amplify.JSONValue.string("ca-central-1"), "PoolId": Amplify.JSONValue.string("redacted"), "AppClientId": Amplify.JSONValue.string("aa")])]), "IdentityManager": Amplify.JSONValue.object(["Default": Amplify.JSONValue.array([])]), "UserAgent": Amplify.JSONValue.string("aws-amplify/cli"), "AppSync": Amplify.JSONValue.object(["Default": Amplify.JSONValue.object(["Region": Amplify.JSONValue.string("ca-central-1"), "ApiUrl": Amplify.JSONValue.string("https://<redacted>.appsync-api.ca-central-1.amazonaws.com/graphql"), "AuthMode": Amplify.JSONValue.string("AMAZON_COGNITO_USER_POOLS"), "ClientDatabasePrefix": Amplify.JSONValue.string("MyAppmanager_AMAZON_COGNITO_USER_POOLS")])]), "Version": Amplify.JSONValue.string("0.1.0"), "CredentialsProvider": Amplify.JSONValue.object(["CognitoIdentity": Amplify.JSONValue.object(["Default": Amplify.JSONValue.object(["Region": Amplify.JSONValue.string("ca-central-1"), "PoolId": Amplify.JSONValue.string("redacted")])])]), "Auth": Amplify.JSONValue.object(["Default": Amplify.JSONValue.object(["authenticationFlowType": Amplify.JSONValue.string("USER_SRP_AUTH"), "OAuth": Amplify.JSONValue.object(["WebDomain": Amplify.JSONValue.string("<redacted>.ca-central-1.amazoncognito.com"), "AppClientId": Amplify.JSONValue.string("aa"), "SignInRedirectURI": Amplify.JSONValue.string("<redacted>/app"), "Scopes": Amplify.JSONValue.array([Amplify.JSONValue.string("email"), Amplify.JSONValue.string("profile"), Amplify.JSONValue.string("openid"), Amplify.JSONValue.string("aws.cognito.signin.user.admin")]), "SignOutRedirectURI": Amplify.JSONValue.string("<redacted>/")])])]), "MyAppmanager_AWS_IAM": Amplify.JSONValue.object(["ApiUrl": Amplify.JSONValue.string("https://<redacted>.appsync-api.ca-central-1.amazonaws.com/graphql"), "AuthMode": Amplify.JSONValue.string("AWS_IAM"), "ClientDatabasePrefix": Amplify.JSONValue.string("MyAppmanager_AWS_IAM"), "Region": Amplify.JSONValue.string("ca-central-1")])])])), dataStore: nil, geo: nil, hub: nil, logging: nil, notifications: nil, predictions: nil, storage: nil))
2023-09-03 14:42:36.554601-0500 My App[60565:14036310] [AuthenticationAWSCognitoAuthPlugin] Credential Store state change:

notConfigured
2023-09-03 14:42:36.554698-0500 My App[60565:14036309] [AuthenticationAWSCognitoAuthPlugin] Auth state change:

{
    "AuthState.notConfigured" =     {
    };
}
2023-09-03 14:42:36.555871-0500 My App[60565:14036309] [AuthenticationAWSCognitoAuthPlugin] Auth state change:

{
    "AuthState.configuringAuth" =     {
    };
}
2023-09-03 14:42:36.556485-0500 My App[60565:14036309] [AuthenticationAWSCognitoAuthPlugin] AWSCognitoAuthPlugin/InitializeAuthConfiguration.swift Starting execution
2023-09-03 14:42:36.557636-0500 My App[60565:14036314] [AuthenticationAWSCognitoAuthPlugin] Credential Store state change:

migratingLegacyStore
2023-09-03 14:42:36.557926-0500 My App[60565:14036310] [AuthenticationAWSCognitoAuthPlugin] AWSCognitoAuthPlugin/MigrateLegacyCredentialStore.swift Starting execution
2023-09-03 14:42:36.559349-0500 My App[60565:14036316] [API<AWSAPIPlugin.AWSAPIPlugin: 0x600000b38a80>] Configure finished
Amplify Initialized
2023-09-03 14:42:36.562270-0500 My App[60565:14036316] [AuthenticationAWSAuthFetchSessionTask] Starting execution
2023-09-03 14:42:36.562963-0500 My App[60565:14036316] [AuthenticationAWSAuthTaskHelper] Check if authstate configured
2023-09-03 14:42:36.567341-0500 My App[60565:14036310] [AuthenticationAWSCognitoAuthPlugin] AWSCognitoAuthPlugin/MigrateLegacyCredentialStore.swift Sending event CredentialStoreEvent.loadCredentialStore
2023-09-03 14:42:36.567569-0500 My App[60565:14036316] [AuthenticationAWSCognitoAuthPlugin] Credential Store state change:

loadingStoredCredentials
2023-09-03 14:42:36.567575-0500 My App[60565:14036314] [AuthenticationAWSCognitoAuthPlugin] AWSCognitoAuthPlugin/LoadCredentialStore.swift Starting execution
2023-09-03 14:42:36.568241-0500 My App[60565:14036314] [AuthenticationAWSCognitoAuthPlugin] AWSCognitoAuthPlugin/LoadCredentialStore.swift Retreiving credential amplifyCredentials
2023-09-03 14:42:36.573880-0500 My App[60565:14036314] [AuthenticationAWSCognitoAuthPlugin] AWSCognitoAuthPlugin/LoadCredentialStore.swift Sending event CredentialStoreEvent.completedOperation
2023-09-03 14:42:36.573950-0500 My App[60565:14036310] [AuthenticationAWSCognitoAuthPlugin] AWSCognitoAuthPlugin/IdleCredentialStore.swift Starting execution
2023-09-03 14:42:36.573965-0500 My App[60565:14036314] [AuthenticationAWSCognitoAuthPlugin] AWSCognitoAuthPlugin/InitializeAuthConfiguration.swift Sending event AuthEvent.validateCredentialAndConfiguration
2023-09-03 14:42:36.573989-0500 My App[60565:14036310] [AuthenticationAWSCognitoAuthPlugin] AWSCognitoAuthPlugin/IdleCredentialStore.swift Sending event CredentialStoreEvent.moveToIdleState
2023-09-03 14:42:36.574037-0500 My App[60565:14036310] [AuthenticationAWSCognitoAuthPlugin] AWSCognitoAuthPlugin/ValidateCredentialsAndConfiguration.swift Starting execution
2023-09-03 14:42:36.574047-0500 My App[60565:14036295] [AuthenticationAWSCognitoAuthPlugin] Auth state change:

{
    "AuthState.validatingCredentialsAndConfiguration" =     {
    };
}
2023-09-03 14:42:36.574079-0500 My App[60565:14036310] [AuthenticationAWSCognitoAuthPlugin] AWSCognitoAuthPlugin/ValidateCredentialsAndConfiguration.swift Sending event AuthEvent.configureAuthentication
2023-09-03 14:42:36.574472-0500 My App[60565:14036295] [AuthenticationAWSCognitoAuthPlugin] Auth state change:

{
    "AuthState.configuringAuthentication" =     {
        "AuthenticationState.notConfigured" =         {
        };
    };
}
2023-09-03 14:42:36.574884-0500 My App[60565:14036316] [AuthenticationAWSCognitoAuthPlugin] Credential Store state change:

success(AWSCognitoAuthPlugin.CredentialStoreData.amplifyCredentials(userPoolAndIdentityPool))
2023-09-03 14:42:36.574914-0500 My App[60565:14036310] [AuthenticationAWSCognitoAuthPlugin] AWSCognitoAuthPlugin/InitializeAuthenticationConfiguration.swift Starting execution
2023-09-03 14:42:36.574922-0500 My App[60565:14036316] [AuthenticationAWSCognitoAuthPlugin] Credential Store state change:

idle
2023-09-03 14:42:36.574953-0500 My App[60565:14036310] [AuthenticationAWSCognitoAuthPlugin] AWSCognitoAuthPlugin/InitializeAuthenticationConfiguration.swift Sending event AuthenticationEvent.configure
2023-09-03 14:42:36.575718-0500 My App[60565:14036310] [AuthenticationAWSCognitoAuthPlugin] AWSCognitoAuthPlugin/ConfigureAuthentication.swift Start execution
2023-09-03 14:42:36.575719-0500 My App[60565:14036316] [AuthenticationAWSCognitoAuthPlugin] Auth state change:

{
    "AuthState.configuringAuthentication" =     {
        "AuthenticationState.configured" =         {
        };
    };
}
2023-09-03 14:42:36.575743-0500 My App[60565:14036310] [AuthenticationAWSCognitoAuthPlugin] AWSCognitoAuthPlugin/ConfigureAuthentication.swift Sending event AuthenticationEvent.initializedSignedIn
2023-09-03 14:42:36.575792-0500 My App[60565:14036310] [AuthenticationAWSCognitoAuthPlugin] AWSCognitoAuthPlugin/ConfigureAuthentication.swift Sending event AuthEvent.authenticationConfigured
2023-09-03 14:42:36.576292-0500 My App[60565:14036310] [AuthenticationAWSCognitoAuthPlugin] AWSCognitoAuthPlugin/InitializeAuthorizationConfiguration.swift Starting execution
2023-09-03 14:42:36.576361-0500 My App[60565:14036310] [AuthenticationAWSCognitoAuthPlugin] AWSCognitoAuthPlugin/InitializeAuthorizationConfiguration.swift Sending event AuthorizationEvent.cachedCredentialsAvailable
2023-09-03 14:42:36.577838-0500 My App[60565:14036337] [AuthenticationAWSCognitoAuthPlugin] AWSCognitoAuthPlugin/ConfigureAuthorization.swift Starting execution
2023-09-03 14:42:36.577874-0500 My App[60565:14036337] [AuthenticationAWSCognitoAuthPlugin] AWSCognitoAuthPlugin/ConfigureAuthorization.swift Sending event AuthEvent.authorizationConfigured
2023-09-03 14:42:36.578274-0500 My App[60565:14036314] [AuthenticationAWSAuthTaskHelper] Auth state configured
2023-09-03 14:42:36.578883-0500 My App[60565:14036316] [AuthenticationAWSCognitoAuthPlugin] Auth state change:

{
    "AuthState.configuringAuthentication" =     {
        "AuthenticationState.signedIn" =         {
            deviceMetadata = "[\"noData\": \"noData\"]";
            signInMethod = "AWSCognitoAuthPlugin.SignInMethod.apiBased(AWSCognitoAuthPlugin.AuthFlowType.userSRP)";
            signedInDate = "2023-09-03 19:29:01 +0000";
            tokens = "[\"expiry\": 2023-09-03 20:29:01 +0000, \"accessToken\": \"ey*****rg\", \"idToken\": \"ey*****yA\", \"refreshToken\": \"ey*****ZQ\"]";
            userId = "ba********************************14";
            userName = "ba********************************14";
        };
    };
}
2023-09-03 14:42:36.579163-0500 My App[60565:14036314] [AuthenticationFetchAuthSessionOperationHelper] Fetching current state
2023-09-03 14:42:36.579174-0500 My App[60565:14036316] [AuthenticationAWSCognitoAuthPlugin] Auth state change:

{
    "AuthState.configuringAuthorization" =     {
        "AuthenticationState.signedIn" =         {
            deviceMetadata = "[\"noData\": \"noData\"]";
            signInMethod = "AWSCognitoAuthPlugin.SignInMethod.apiBased(AWSCognitoAuthPlugin.AuthFlowType.userSRP)";
            signedInDate = "2023-09-03 19:29:01 +0000";
            tokens = "[\"refreshToken\": \"ey*****ZQ\", \"idToken\": \"ey*****yA\", \"accessToken\": \"ey*****rg\", \"expiry\": 2023-09-03 20:29:01 +0000]";
            userId = "ba********************************14";
            userName = "ba********************************14";
        };
        "AuthorizationState.notConfigured" =         {
        };
    };
}
2023-09-03 14:42:36.579188-0500 My App[60565:14036314] [AuthenticationFetchAuthSessionOperationHelper] Session exists, checking validity
2023-09-03 14:42:36.579363-0500 My App[60565:14036316] [AuthenticationAWSCognitoAuthPlugin] Auth state change:

{
    "AuthState.configuringAuthorization" =     {
        "AuthenticationState.signedIn" =         {
            deviceMetadata = "[\"noData\": \"noData\"]";
            signInMethod = "AWSCognitoAuthPlugin.SignInMethod.apiBased(AWSCognitoAuthPlugin.AuthFlowType.userSRP)";
            signedInDate = "2023-09-03 19:29:01 +0000";
            tokens = "[\"idToken\": \"ey*****yA\", \"accessToken\": \"ey*****rg\", \"expiry\": 2023-09-03 20:29:01 +0000, \"refreshToken\": \"ey*****ZQ\"]";
            userId = "ba********************************14";
            userName = "ba********************************14";
        };
        "AuthorizationState.sessionEstablished" = userPoolAndIdentityPool;
    };
}
2023-09-03 14:42:36.579577-0500 My App[60565:14036316] [AuthenticationAWSCognitoAuthPlugin] Auth state change:

{
    "AuthState.configured" =     {
        "AuthenticationState.signedIn" =         {
            deviceMetadata = "[\"noData\": \"noData\"]";
            signInMethod = "AWSCognitoAuthPlugin.SignInMethod.apiBased(AWSCognitoAuthPlugin.AuthFlowType.userSRP)";
            signedInDate = "2023-09-03 19:29:01 +0000";
            tokens = "[\"accessToken\": \"ey*****rg\", \"refreshToken\": \"ey*****ZQ\", \"expiry\": 2023-09-03 20:29:01 +0000, \"idToken\": \"ey*****yA\"]";
            userId = "ba********************************14";
            userName = "ba********************************14";
        };
        "AuthorizationState.sessionEstablished" = userPoolAndIdentityPool;
    };
}
```
</details>
```


### Is this a regression?

No

### Regression additional context

_No response_

### Platforms

iOS

### OS Version

16.4

### Device

iPhone 14 Pro Max Simulator

### Specific to simulators

_No response_

### Additional context

_No response_

**Comment by vmo-bstock** [comment]
So, after much...much more head scratching, I tried to use OIDC auth for the REST API as described here: 
[https://docs.amplify.aws/lib/restapi/authz/q/platform/ios/#oidc](url)

And, during my testing, I noted that the "AccessToken" wasn't able to authenticate against the COGNITO POOL, but the idToken could, so I followed what was described here: 
[https://docs.amplify.aws/lib/restapi/authz/q/platform/ios/#note-related-to-use-access-token-or-id-token](url) 
to override the default behaviour of using the AccessToken for requests and chose the idToken.  Everything is working as expected now.  

Is there some suggestions as to why the default "AccessToken" wasn't working but the idTokens are?  😣
